File: java/src/tarehart/rlbot/JumpingBeanBot.java
package tarehart.rlbot;

import tarehart.rlbot.input.CarData;
import tarehart.rlbot.planning.Plan;
import tarehart.rlbot.planning.SetPieces;
import tarehart.rlbot.planning.TacticsAdvisor;
import tarehart.rlbot.time.GameTime;

import java.util.Optional;

public class JumpingBeanBot extends Bot {

    private TacticsAdvisor tacticsAdvisor;

    public JumpingBeanBot(Team team, int playerIndex) {
        super(team, playerIndex);
    }

    private Plan plan;

    @Override
    protected AgentOutput getOutput(AgentInput input) {

        if (!Plan.Companion.activePlan(plan).isPresent()) {
            plan = SetPieces.jumpSuperHigh(10);
        }

        return plan.getOutput(input).get();
    }
}

File: java/src/tarehart/rlbot/JumpingBeanBot.kt
package tarehart.rlbot

import tarehart.rlbot.input.CarData
import tarehart.rlbot.planning.Plan
import tarehart.rlbot.planning.SetPieces
import tarehart.rlbot.planning.TacticsAdvisor
import tarehart.rlbot.time.GameTime

import java.util.Optional

class JumpingBeanBot(team: Bot.Team, playerIndex: Int) : Bot(team, playerIndex) {

    override fun getOutput(input: AgentInput): AgentOutput {

        if (Plan.activePlanKt(currentPlan) == null) {
            currentPlan = SetPieces.jumpSuperHigh(10.0)
        }

        return currentPlan?.getOutput(input) ?: AgentOutput()
    }
}
